daemon/reload.go

```diff
@@ -24,6 +24,9 @@ import (
 // - Registry mirrors
 // - Daemon live restore
 func (daemon *Daemon) Reload(conf *config.Config) (err error) {
+	if err := verifyDaemonSettings(conf); err != nil {
```

Comment:
`conf` may be a partial config (with zeroes in place of defaults) and so may not pass verification

Reply:
I reverted this and just moved the shim name check to the config package.